File: SPRING/4-angular-products/src/app/services/product.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { map, Observable } from 'rxjs';
import { Product } from '../models/product';

@Injectable({
  providedIn: 'root'
})
export class ProductService {

  private url: string = "http://localhost:8080"

  constructor(private http: HttpClient) { }

  findAll(): Observable<Product[]>{
    return this.http.get(this.url).pipe(
      map((response: any ) => response as Product[])
    );
  }

  create(product: Product): Observable<Product>{
    return this.http.post<Product>(this.url, product);
  }

  update(product: Product): Observable<Product>{
    return this.http.put<Product>(`${this.url}/${product.id}`, product);
  }

  remove(id: number): Observable<Product>{
    return this.http.delete<Product>(`${this.url}/${id}`);
  }

}

File: Master de Java/FX/javafx-app/src/main/java/org/sam/javafx/app/javafxapp/services/ProductService.java
package org.sam.javafx.app.javafxapp.services;

import org.sam.javafx.app.javafxapp.models.Product;

import java.util.List;

public interface ProductService {

    List<Product> findAll();
    Product save(Product product);
    Product update(Product product);
    Product delete(Product product);

}
